pkg/controller/compliancesuite/suitererunner_cron_compat.go

```diff
@@ -0,0 +1,254 @@
+package compliancesuite
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	compv1alpha1 "github.com/ComplianceAsCode/compliance-operator/pkg/apis/compliance/v1alpha1"
+	"github.com/ComplianceAsCode/compliance-operator/pkg/controller/common"
+	"github.com/ComplianceAsCode/compliance-operator/pkg/utils"
+)
+
+// GetRerunnerName gets the name of the rerunner workload based on the suite name
+func GetRerunnerName(suiteName string) string {
+	// Operator SDK doesn't allow CronJob with names longer than 52
+	// characters. Trim everything but the first 42 characters so we have
+	// enough room for the "-rerunner" string.
+	if len(suiteName) >= 42 {
+		suiteName = suiteName[0:42]
+	}
+	return suiteName + "-rerunner"
+}
+
+func (r *ReconcileComplianceSuite) cronJobCompatCreate(
+	suite *compv1alpha1.ComplianceSuite,
+	key types.NamespacedName,
+	logger logr.Logger,
+) error {
+	var getObj client.Object
+
+	priorityClassName, err := r.getPriorityClassName(suite)
+	if err != nil {
+		logger.Error(err, "Cannot get priority class name, scan will not be run with set priority class")
```

Comment:
Future logging improvement, we could include the scan/suite name and the priority class in this message to help operators trace things.

Something to consider for a follow up.